plugin/fuzz/source/dextool/plugin/fuzz/sutxml.d

```diff
@@ -0,0 +1,208 @@
+/*
+    This will be changed to something else. This file should be considered a POC until this changes. 
+TODO: Should be more modular.
+*/
+
+module sutxml;
+
+import std.string;
+import std.file;
+import std.path;
+import std.algorithm;
+import std.stdio;
+import std.array;
+import std.container.array;
+import std.xml;
+
+//string[] interfaces =  ["ContinuesInterface"]; //Monitored and Event should probl. be added here
+
+struct XMLType {
+    this(string name, string type) {
+        this.name = name;
+        this.type = type;
+    }
+
+    string name;
+    string type;
+    string unit; ///Optional
+    string min; ///Optional
+    string max; ///Optional
+}
+
+//Generic interface struct(?)
+struct Interface {
+    ContinuesInterface ci;
+    //Add more interfaces here
+    alias ci this;
+    //Add more alias here
+}
+
+struct ContinuesInterface {
+    int direction; //From = 0; To = 1;
+    auto data_items = Array!DataItem();
+}
+
+struct DataItem {
+
+    string name;
+    string type;
+
+    string startupVal;
+    string defaultVal;
+    this(string name, string type) {
+        this.name = name;
+        this.type = type;
+    }
+}
+
+@safe struct SutEnvironment {
+private:
+    Document doc;
+    string basedir;
+    string[] foundNamespaces;
+    string[] xmlinterfaces;
+    Array!XMLType[string] xmltypes;
+
+    string namespace(string filename) {
+        ///TODO: Namespaces should probably only be added if there is any interface in the file
+        //dfmt off
+        if (!initDocument(filename) || !checkInterface(filename)) {
+            return "";
+        }
+        return dirName(filename)
+                .chompPrefix(this.basedir)
+                .replace(dirSeparator, "::")
+                .chompPrefix("::")
+                ~ "::" ~ baseName(filename, ".xml");
+            //dfmt on
+    }
+
+    @trusted Array!XMLType getAllSubtypes(Element types) { //Types = Types tag
+        Array!XMLType xml_types;
+        foreach (Element subtype; types.elements) {
+            XMLType curr_type = XMLType(subtype.tag.attr["name"], subtype.tag.attr["type"]);
+
+            if (subtype.tag.attr.get("min", "-1NOTFOUND") != "-!NOTFOUND") {
```

Comment:
If I see correctly attr is an associative array.
The following pattern ensures only one lookup is made in the AA:
```d
string[string] arr = ["a": "foo", "b":"borf"];
if (auto v = "a" in arr) {
    writeln(*v);
}
```